articles/what-is-jenkins/index.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/what-is-jenkins/
+title: What is Jenkins
+description: Developers use Jenkins to continuously build and test software projects, making it easier to integrate changes to the project and making it easier for users to obtain a good build with no errors and bugs.
+author: judy-nduati
+date: 2020-09-15T00:00:00-10:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/what-is-jenkins/hero.jpg
+    alt: Jenkins Continuous Integration tool
+---
+Jenkins is a Continuous Integration (CI) tool and an open-source tool written in Java. Jenkins achieves Continuous Integration with the help of plugins, which are over a thousand in count. Developers mostly use Jenkins for Continuous Integration because of its flexibility and the number of plugins it supports.
+<!--more-->
+### What is Jenkins?
+Developers use Jenkins to continuously build and test software projects, making it easier to integrate changes to the project, and making it easier for users to obtain a good build with no errors and bugs. Jenkins allows developers to continuously deliver a software project by integrating with a large number of development, deployment, and testing technologies such as Git, Maven, Puppet, Selenium, Ansible, and Nagios.
+
+### Problems Developers faced before Continuous Integration
+Earlier before Continuous Integration was introduced, a team of developers used to work from different locations, each of them would individually implement a few classes. They would write code and tests to ensure that their classes work properly. Then assume the software project will build and run smoothly because classes are tested well. They then integrate the classes and combine them to create a larger program that was being developed by the team, but suddenly everything breaks.  There are issues with integration, the code fails to compile, all sorts of bugs arise, and things stop working out. This is also called [integration hell](https://www.solutionsiq.com/agile-glossary/integration-hell/#:~:text=Integration%20Hell%20refers%20to%20the,that%20it%20can%20finally%20integrate.)
+Integration hell is a major problem to software projects and it is risky to their success reason. Seeing it was hard to estimate how long it would take to fix the things that went wrong. Developers had to figure out what went wrong, and that could be very hard to do in a large project. After diagnosing the problems, again time had to be spent fixing them, the project budget may also increase and the success of the project is risked. The solution to integration hell was to integrate continuously throughout the entire project, and thus the practice of Continuous Integration was introduced to the DevOps field.
```

Comment:
Rephrase topic sentence to make sense. What is meant by integrate continuously? Clarify how continuous integration allows developers not only to work together remotely, but merge their code in a safe way by referencing a source or using more descriptive detail.